### PR TITLE
fix: allow adding self to shared event

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
Resolves: https://github.com/nextcloud/calendar/issues/2659

Modified alreadyInvitedEmails result to NOT exclude the current user by default and to only include actual organize and attendees.

### CODE REVIEW

**Review by st3iny — CHANGES_REQUESTED** [review]
Breaks the attendees tab on new events: `TypeError: Cannot read properties of null (reading 'uri')`.

File: src/components/Editor/Invitees/InviteesList.vue [hunk 1]
```diff
@@ -255,9 +255,8 @@ export default {
 		alreadyInvitedEmails() {
 			const emails = this.invitees.map(attendee => removeMailtoPrefix(attendee.uri))
 
-			const principal = this.principalsStore.getCurrentUserPrincipal
-			if (principal) {
-				emails.push(principal.emailAddress)
+			if (this.calendarObjectInstance.organizer.uri) {
+				emails.push(removeMailtoPrefix(this.calendarObjectInstance.organizer.uri))
```
Comment (st3iny) [inline src/components/Editor/Invitees/InviteesList.vue:259]:
```suggestion
			// A user should be able to invite themselves if they are not the organizer
			const principal = this.principalsStore.getCurrentUserPrincipal
			const organizerUri = this.calendarObjectInstance.organizer?.uri
			if (organizerUri && removeMailtoPrefix(organizerUri) !== principal.emailAddress) {
				emails.push(removeMailtoPrefix(organizerUri))
			} else if (principal) {
				emails.push(principal.emailAddress)
			}
```

Only add the organizer's URI if it isn't equal to the current principal's email address. I should only be able to invite myself if I'm not the organizer.

This was tested locally. See my other comment above.

**Comment by st3iny** [comment]
Do we want to backport this (since the issue is from 2020)? Or is this too risky because of the recent shared calendar changes in server?

**Comment by SebastianKrupinski** [comment]
> Do we want to backport this (since the issue is from 2020)? Or is this too risky because of the recent shared calendar changes in server?

We can back port this to 30 for sure and maybe all the way to 28, but I have to look at the code in 29/28 first, it's probably safe just want to make sure.

**Pull request merged**